core/src/test/java/org/elasticsearch/indices/state/OpenCloseIndexIT.java

```diff
@@ -190,6 +190,39 @@ public void testCloseOpenAllWildcard() {
         assertThat(openIndexResponse.isAcknowledged(), equalTo(true));
         assertIndexIsOpened("test1", "test2", "test3");
     }
+    
+    // if there are no indices to open/close throw an exception
+    public void testOpenCloseWildCardsNoIdcicesDefault() {
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("test").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("test").execute().actionGet());
+        
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("test*").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("test*").execute().actionGet());
+        
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("*").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("*").execute().actionGet());
+        
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("_all").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("_all").execute().actionGet());
+    }
+    
+    // if there are no indices to open/close and allow_no_indices=true, the open/close is a no-op
+    public void testOpenCloseWildCardsNoIdcicesAllowNoIndices() {
+        IndicesOptions openIncidesOptions = IndicesOptions.fromOptions(false, true, false, true);
+        IndicesOptions closeIncidesOptions = IndicesOptions.fromOptions(false, true, true, false);
+
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareOpen("test").execute().actionGet());
+        expectThrows(IndexNotFoundException.class, () -> client().admin().indices().prepareClose("test").execute().actionGet());
```

Comment:
this looks like the same tests that you have in the other method. Did you mean to set the indices options here too? if you set ignore unavailable to true even these calls are acked with no exception right?